plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployJob.java

```diff
@@ -133,28 +112,54 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
         return saveStatus;
       }
 
-      IStatus stagingStatus = stageProject(credentialFile, stagingDirectory, progress.newChild(30));
+      synchronized (this) {
+        if (canceled) {
+          return Status.CANCEL_STATUS;
+        }
+        deployCloudSdkFacade = CloudSdkProcessFacade.forDeploy(credentialFile, stderrLineListener);
+        stagingCloudSdkFacade = CloudSdkProcessFacade.forStandardStaging(getProjectVm(project),
```

Comment:
Odd to have `forStandardStaging()` when this could be for flexible.

Perhaps both of these are moved into the appropriate stager/deployer?